### DISCUSSION TITLE
How to add ImageButton to custom coordinates?

### DISCUSSION THREAD

**Discussion by breadfan** [emilk/egui#2499] [Q&A]
Connected with previous question, I have some canvas/image, where certain parts (circles) are need to be handled as ImageButtons. Can I do it without making whole canvas UserInterface object?

**Comment by John-Nagle** [comment]
I need that too. I need to make game controls like this:
![arrowbuttons](https://user-images.githubusercontent.com/942608/210285512-233a29f4-9cb3-4e8c-b598-a1c3f287b693.png)

I suppose I'll have to get the hit UV values and hard code the geometry.

**Reply by breadfan** [reply]
There's no way to put an ImageButton to the custom place. 
In your case you could use `macroquad` to get user clicks and to draw primitives like triangles, rectangles, circles.

**Comment by John-Nagle** [comment]
Just wrote a widget to do a 4-direction button like a game controller. You can get the click position and the button rect in a Result, so there's enough info passed through to do this easily. 

The example for "ui-mock" uses it. Not merged to main yet, "navigation-buttons" branch only. 

https://github.com/John-Nagle/ui-mock/blob/navigation-buttons/src/navarrows.rs

**Reply by breadfan** [reply]
You just wrote a custom widget. What was my question about?
